src/main/java/io/fabric8/jenkins/openshiftsync/ImageStreamWatcher.java

```diff
@@ -0,0 +1,199 @@
+                        final ImageStreamList imageStreams = getAuthenticatedOpenShiftClient().imageStreams().inNamespace(namespace).list();
+                        onInitialImageStream(imageStreams);
+                        logger.fine("handled ImageStream resources");
+                        if (watches.get(namespace) == null) {
+                            watches.put(namespace,getAuthenticatedOpenShiftClient().imageStreams().inNamespace(namespace).withResourceVersion(imageStreams.getMetadata().getResourceVersion()).watch(ImageStreamWatcher.this));
+                        }
+                    } catch (Exception e) {
+                        logger.log(SEVERE, "Failed to load ImageStreams: " + e, e);
+                    }
+                }
+            }
+        };
+    }
+
+    public synchronized void start() {
+        // lets process the initial state
+        logger.info("Now handling startup image streams!!");
+        super.start();
+    }
+
+    @Override
+    public void eventReceived(Action action, ImageStream imageStream) {
+        try {
+            Map<String,PodTemplate> slavesFromIS = podTemplates(imageStream);
+            String isuuid = imageStream.getMetadata().getUid();
+            switch (action) {
+                case ADDED:
+                    for (Map.Entry<String,PodTemplate> entry : slavesFromIS.entrySet()) {
+                        trackedImageStreams.put(entry.getKey(), entry.getValue());
+                        JenkinsUtils.addPodTemplate(entry.getValue());
+                    }
+                    break;
+
+                case MODIFIED:
+                    // add/replace entries from latest IS incarnation
+                    for (Map.Entry<String,PodTemplate> entry : slavesFromIS.entrySet()) {
+                        boolean alreadyTracked = trackedImageStreams.containsKey(entry.getKey());
+                        if (alreadyTracked) {
+                            JenkinsUtils.removePodTemplate(trackedImageStreams.get(entry.getKey()));
+                        }
+                        JenkinsUtils.addPodTemplate(entry.getValue());
+                        trackedImageStreams.put(entry.getKey(), entry.getValue());
+                    }
+                    // go back and remove tracked items that no longer are marked slaves
+                    Set<String> keyset = trackedImageStreams.keySet();
+                    for (String key : keyset) {
+                        // if the entry is not for the IS from the watch event, skip
+                        if (!key.startsWith(isuuid))
+                            continue;
+                        if (!slavesFromIS.containsKey(key)) {
+                            JenkinsUtils.removePodTemplate(trackedImageStreams.get(key));
+                            keyset.remove(key); // removes from trackedImageStreams as well
+                        }
+                    }
+                    break;
+
+                case DELETED:
```

Comment:
yep, agreed - actually meant to make the same change there last night, but got distracted and forgot  .... will push in a bit